server/jetstream_events.go

```diff
@@ -193,10 +193,21 @@ const JSRestoreCompleteAdvisoryType = "io.nats.jetstream.advisory.v1.restore_com
 
 // Clustering specific.
 
-// JSStreamLeaderElectedAdvisoryType is sent when the system elects a leader for a stream.
+// JSClusterLeaderElectedAdvisoryType is sent when the system elects a new meta leader.
+const JSDomainLeaderElectedAdvisoryType = "io.nats.jetstream.advisory.v1.domain_leader_elected"
+
+// JSClusterLeaderElectedAdvisory indicates that a domain has elected a new leader.
+type JSDomainLeaderElectedAdvisory struct {
+	TypedEvent
+	Leader   string      `json:"leader"`
+	Replicas []*PeerInfo `json:"replicas"`
+	Domain   string      `json:"domain,omitempty"`
```

Comment:
Maybe add `cluster`?

Reply:
Not following that, what do you mean?

Reply:
It would be good to know what cluster hosts the leader in this event.

Reply:
stream is where it is, it'll never change until someone goes and specifically moves a stream so its not needed there I think.

But the leader moves around and one might want to react on the advisory should the leader move to a bad place, be good if the info needed to answer if its a bad place is all in the advisory.

Reply:
ok done.